src/main/java/net/snowflake/client/jdbc/cloud/storage/EncryptionProvider.java

```diff
@@ -96,10 +96,7 @@ public static void decrypt(
       SecretKey queryStageMasterKey = new SecretKeySpec(qsmkBytes, 0, qsmkBytes.length, AES);
       keyCipher.init(Cipher.DECRYPT_MODE, queryStageMasterKey);
       byte[] fileKeyBytes = keyCipher.doFinal(keyBytes);
-
-      // NB: we assume qsmk.length == fileKey.length
-      //     (fileKeyBytes.length may be bigger due to padding)
-      fileKey = new SecretKeySpec(fileKeyBytes, 0, qsmkBytes.length, AES);
+      fileKey = new SecretKeySpec(fileKeyBytes, AES);
```

Comment:
is this affect all cloud providers? What will happen when fileKeyBytes.length is bigger due to padding?

Reply:
Yes, this would affect all cloud providers. The padding is taken care of by the Cipher. When we initialize it with Cipher.getInstance(KEY_CIPHER), it is being initialized with AES/ECB/PKCS5Padding. When the fileKey is constructed, normally there is no issue as qsmk and fileKey are the same length, and no important key information is truncated. But in GS, we sometimes encrypt with AES 256, which has a keysize of 32 bytes, meaning if it is decrypted here, we truncate 16 bytes of the key, leading to a padding exception. This update matches our GS code [here](https://github.com/snowflakedb/snowflake/blob/6f80bd3a71157af95e4a9bc2681c3b3dded5f81a/GlobalServices/modules/dbsec/crypto-api/src/main/java/com/snowflake/security/Cryptor.java#L549).

Reply:
You can see the bug in the linked JIRA description.